saleor/graphql/core/types/sort_input.py

```diff
@@ -0,0 +1,21 @@
+import graphene
+
+from ..enums import OrderDirection
+
+
+class SortInputObjectType(graphene.InputObjectType):
+    direction = graphene.Argument(
+        OrderDirection,
+        required=True,
+        description="Specifies the direction in which to sort products.",
+    )
+
+    @classmethod
+    def __init_subclass_with_meta__(
+        cls, container=None, _meta=None, sort_enum=None, type_name=None, **options
+    ):
+        super().__init_subclass_with_meta__(container, _meta, **options)
+        field = graphene.Argument(
+            sort_enum, description=f"Sort {type_name} by the selected field."
```

Comment:
This field should be required in the schema. The only exception is `ProductOrder` which also allows sorting by attributes - there both fields should be optional. If it's possible we should try to make this field required here and maybe override it in `ProductOrder` and make it optional there.